test/com/linkedin/parseq/TestEngine.java

```diff
@@ -148,6 +148,33 @@ public void testShutdownWithRunningAndSuccessorTask() throws InterruptedExceptio
   }
 
   @Test
+  public void testShutdownWithSideEffectTask() throws InterruptedException {
+    final CountDownLatch finishLatch = new CountDownLatch(1);
+    final String mainValue = "main task executed";
+    final String sideEffectValue = "side-effect task executed";
+
+    Task<String> sideEffect = Task.async(context -> {
+      finishLatch.await();
```

Comment:
Could you also add a test that is exactly the same except that it does not block a parseq thread but instead returns uncompleted promise?